Common.mk

```diff
@@ -307,6 +323,106 @@ define CGO_BINARY_TARGET_BODY
 
 endef
 
+define GO_MOD_DOWNLOAD_ALL_TARGETS_BODY	
+	$(eval $(foreach path, $(call uniq,$(GO_MOD_PATHS)), $(call GO_MOD_DOWNLOAD_TARGET_BODY,$(path))))
+endef
+
+# $1 - go mod path for binary
+define GO_MOD_DOWNLOAD_TARGET_BODY
+	$(REPO)/$(1)/eks-distro-go-mod-download: $(if $(PATCHES_DIR),$(GIT_PATCH_TARGET),$(GIT_CHECKOUT_TARGET))
+		$$(BASE_DIRECTORY)/build/lib/go_mod_download.sh $$(MAKE_ROOT) $$(REPO) $$(GIT_TAG) $$(GOLANG_VERSION) $(1)
+		@touch $$@
+
+endef
+
+# $(eval $(if $(strip $(filter-out .,$(GO_MOD_PATHS))),$(call pairmap,GATHER_LICENSES_TARGETS_BODY,$(BINARY_TARGET_FILES),$(GO_MOD_PATHS)),$(call GATHER_LICENSES_TARGETS_BODY,.)))
+# If go_mod_paths is all `.` then only create the one target for the default /attribution folder for licenses
+# otherwise for each go mod path, generate a new license target
+# $1 - binary file name
+# $2 - go mod path for binary
+# Strange make issue here, calling $(eval $(call ...)) directly will not work as intended
+# "looping" over the list of 1 to avoid the issue
+define GATHER_LICENSES_ALL_TARGETS_BODY	
+	$(eval $(foreach binary,$(1), $(call GATHER_LICENSES_TARGETS_BODY,$(binary),$(2))))
+endef
+
+# $1 - binary file name
+# $2 - go mod path for binary
+define GATHER_LICENSES_TARGETS_BODY
+	$(call LICENSE_TARGET_FROM_BINARY_GO_MOD,$(1),$(2)): $(REPO)/$(2)/eks-distro-go-mod-download
+		$$(BASE_DIRECTORY)/build/lib/gather_licenses.sh $$(REPO) $(MAKE_ROOT)/$(call LICENSE_OUTPUT_FROM_BINARY_GO_MOD,$(1),$(2)) "$$(LICENSE_PACKAGE_FILTER)" $(2)
+
```

Comment:
do we need this extra line here?

Reply:
Yeah, not sure why but any of these "functions" which are evald like this need the trailing newline